packages/flutter/lib/src/services/platform_channel.dart

```diff
@@ -143,7 +143,7 @@ class MethodChannel {
   ///   [JSONMethodCodec].
   /// * <https://docs.flutter.io/javadoc/io/flutter/plugin/common/MethodCall.html>
   ///   for how to access method call arguments on Android.
-  Future<dynamic> invokeMethod(String method, [dynamic arguments]) async {
+  Future<T> invokeMethod<T>(String method, [dynamic arguments]) async {
```

Comment:
I don't really like this change. The T doesn't help improve correctness in reality; the method itself completely ignores it, for example. This is similar to the _onEvent case, `dynamic` really is all we can prove.

Reply:
Yeah, this didn't seem to add much except in that one case.  I'll back it out.

Reply:
Backed this out, which requires an additional callback signature change that I called out below.